lib/src/rules/avoid_operator_equals_on_mutable_classes.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/ast.dart';
+
+const _desc =
+    r'AVOID overloading operator == on classes not marked `@immutable`.';
+
+const _details = r'''
+
+**AVOID** overloading operator == on classes not marked `@immutable`.
```

Comment:
This and the `_desc` above should probably be expanded to include hashCode in the list of members being checked for.

Reply:
Done